source/extensions/filters/network/mysql_proxy/mysql_codec_clogin.h

```diff
@@ -25,6 +25,9 @@ class ClientLogin : public MySQLCodec {
   const std::vector<uint8_t>& getAuthResp() const { return auth_resp_; }
   const std::string& getDb() const { return db_; }
   const std::string& getAuthPluginName() const { return auth_plugin_name_; }
+  const std::vector<std::pair<std::string, std::string>> getConnectionAttribute() const {
+    return conn_attr_;
```

Comment:
vector is returned by value, which means all its value will be copied. Consider returning by `reference` or `const reference`.

Reply:
fixed.